tests/main/helpers/test_login_helpers.py

```diff
@@ -0,0 +1,62 @@
+import mock
+import pytest
+
+from collections import namedtuple
+
+from app.main.helpers.login_helpers import is_safe_url
+
+
+@pytest.fixture()
+def host_url(request):
+    fake_request = namedtuple('Request', ['host_url'])('http://localhost/')
+    request_patch = mock.patch('app.main.helpers.login_helpers.request', fake_request)
+    request.addfinalizer(request_patch.stop)
+
+    return request_patch.start()
+
+
+@pytest.fixture(params=['buyer', 'supplier', 'admin', 'admin-ccs-category'])
+def current_user(request):
+    fake_user = namedtuple('User', ['role'])(request.param)
+    request_patch = mock.patch('app.main.helpers.login_helpers.current_user', fake_user)
+    request.addfinalizer(request_patch.stop)
+
+    return request_patch.start()
+
+
+@pytest.mark.parametrize('next_url', [None, ''])
+def test_empty_next_url_is_not_safe(host_url, next_url):
+    assert not is_safe_url(next_url)
+
+
+@pytest.mark.parametrize('next_url', [
+    'https://example.com/suppliers',
+    'http://localhost:5000/',
+    'http://example.com/',
+    '//example.com/',
```

Comment:
Worth adding in the `@example.com` pattern as well (it was mentioned in that article)?

Reply:
That's a greate shout :+1:

I've tried it and it fails the test, but the reason is that the redirect URL ends up being `http://localhost/@example.com`, which is fine. I've added a test to `test_auth.py` for this case.

Reply:
Awesome - looks good!